### PR TITLE
TCVP-2307 Insert email history failing

### PR DESCRIPTION
**Pull request by lfdgov**
# Description

This PR includes the following proposed change(s):

- TCVP-2307

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by KDNeufeld — APPROVED** [review]
Great work

File: src/backend/TrafficCourts/Workflow.Service/Consumers/SendEmailToDisputantConsumer.cs [hunk 1]
```diff
@@ -31,46 +32,62 @@ public SendEmailToDisputantConsumer(IEmailSenderService emailSenderService, IClo
...
 
-        // try to fill in to if missing
-        if (emailMessage.To.IsNullOrEmpty() && dispute is not null && dispute.EmailAddress is not null && dispute.EmailAddressVerified == true)
-            emailMessage.To = dispute.EmailAddress;
+            // try to fill in to if missing
+            if (emailMessage.To.IsNullOrEmpty() && dispute is not null && dispute.EmailAddress is not null && dispute.EmailAddressVerified == true)
+                emailMessage.To = dispute.EmailAddress;
 
-        _logger.LogDebug("Calling email sender service");
+            var result = emailMessage.To.IsNullOrEmpty() ? SendEmailResult.Filtered : await _emailSenderService.SendEmailAsync(emailMessage, context.CancellationToken);
```
Comment (pbolduc) [inline src/backend/TrafficCourts/Workflow.Service/Consumers/SendEmailToDisputantConsumer.cs:51]: This `IsNullOrEmpty()` is from `Microsoft.IdentityModel.Tokens` which checks whether enumerable if character is null or empty. Typically, should be using `string.IsNullOrEmpty(emailMessage.To)`

File: src/backend/TrafficCourts/Workflow.Service/Consumers/DisputantEmailSentConsumer.cs [hunk 2]
```diff
@@ -23,10 +24,24 @@ public DisputantEmailSentConsumer(IOracleDataApiService oracleDataApiService, IL
...
-            await _oracleDataApiService.CreateEmailHistoryAsync(emailHistory, context.CancellationToken);
+                await _oracleDataApiService.CreateEmailHistoryAsync(emailHistory, context.CancellationToken);
+            } 
+            catch (WebException ex)
+            {
+                HttpWebResponse? errorResponse = ex.Response as HttpWebResponse;
+                if (errorResponse?.StatusCode == HttpStatusCode.NotFound || errorResponse?.StatusCode == HttpStatusCode.BadRequest)
+                {
+                    // dont requeue if dispute not found or bad request
+                    _logger.LogError(ex.Message, context);
```
Comment (pbolduc) [inline src/backend/TrafficCourts/Workflow.Service/Consumers/DisputantEmailSentConsumer.cs:41]: This is incorrect usage of this logging method, the signature is `logger.LogError(string? message, params object?[] args)`. There is no format string being passed. Also, do not log `context`. This is a large object.  If you need specific properties, pull them out and add as properties.

File: src/backend/TrafficCourts/Workflow.Service/Consumers/DisputantEmailFilteredConsumer.cs [hunk 3]
```diff
@@ -23,10 +24,26 @@ public DisputantEmailFilteredConsumer(IOracleDataApiService oracleDataApiService
...
+
+                await _oracleDataApiService.CreateEmailHistoryAsync(emailHistory, context.CancellationToken);
+            }
+            catch (WebException ex)
+            {
+                HttpWebResponse? errorResponse = ex.Response as HttpWebResponse;
+                if (errorResponse?.StatusCode == HttpStatusCode.NotFound || errorResponse?.StatusCode == HttpStatusCode.BadRequest)
+                {
+                    // dont requeue if dispute not found or bad request
+                    _logger.LogError(ex.Message, context);
```
Comment (pbolduc) [inline src/backend/TrafficCourts/Workflow.Service/Consumers/DisputantEmailFilteredConsumer.cs:42]: This is incorrect usage of this logging method, the signature is `logger.LogError(string? message, params object?[] args)`. There is no format string being passed. Also, do not log `context`. This is a large object.  If you need specific properties, pull them out and add as properties.

**Review by pbolduc — COMMENTED** [review]
The logging needs to be fixed

**Pull request merged**